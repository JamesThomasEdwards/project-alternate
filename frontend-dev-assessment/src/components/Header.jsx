//React
import React from 'react';

const Header = (props) => {
    // let headerContent = props.headerData.map((ele) => {
    //     return <p className='header__content'>{ele}</p>
    // })
    return (
        <nav>
            <div className='header'>
                <p className='header__content'>{props.headerDataOne}</p>
                <p className='header__content'>{props.headerDataTwo}</p>
                <p className='header__content'>{props.headerDataThree}</p>
                <p className='header__content'
                    onClick={props.toggleHeaderContent}>{props.headerDisplay ? '-' : '+'}
                </p>
            </div>
        </nav>
    )
}


export default Header