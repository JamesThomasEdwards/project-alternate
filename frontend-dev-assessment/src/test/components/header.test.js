// React;
import React from 'react';
// File;
import Header from '../../components/Header.jsx';
// Testing;
import ReactShallowRenderer from 'react-test-renderer/shallow';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

test('should render Header component correctly', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<Header />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
});

describe('Header component test', () => {
    describe('<Header />', () => {
        let wrapper = shallow(<Header headerDataOne='30%'
            headerDataTwo='$2000000'
            headerDataThree='85%'
            toggleHeaderContent={undefined}
            headerDisplay={undefined} />)
        it('Renders a <nav>', () => {
            expect(wrapper.type()).toEqual('nav');
        });
    });
});
