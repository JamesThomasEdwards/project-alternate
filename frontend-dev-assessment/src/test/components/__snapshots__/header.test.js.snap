// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header component correctly 1`] = `
<nav>
  <div
    className="header"
  >
    <p
      className="header__content"
    />
    <p
      className="header__content"
    />
    <p
      className="header__content"
    />
    <div
      className="header__content"
    >
      <button
        className="header__button"
      >
        <p>
          +
        </p>
      </button>
    </div>
  </div>
</nav>
`;
